core/eolearn/tests/test_eodata_io.py

```diff
@@ -611,3 +612,24 @@ def test_partial_saving_fails(eopatch: EOPatch):
         with pytest.raises(ValueError):
             # temporal selection outside of saved eopatch
             eopatch.save(**io_kwargs, temporal_selection=slice(12, None))
+
+
+@pytest.mark.parametrize("patch_location", [".", "patch-folder", "some/long/path"])
+def test_old_style_meta_info(patch_location):
+    with TempFS() as temp_fs:
+        EOPatch(bbox=DUMMY_BBOX).save(path=patch_location, filesystem=temp_fs)
+        meta_info = {"this": ["list"], "something": "else"}
+        with temp_fs.open(f"{patch_location}/meta_info.json", "w") as old_style_file:
+            json.dump(meta_info, old_style_file)
+
+        with pytest.warns(EODeprecationWarning):
+            loaded_patch = EOPatch.load(path=patch_location, filesystem=temp_fs)
+        assert dict(loaded_patch.meta_info.items()) == meta_info
+
+        loaded_patch.meta_info = {"beep": "boop"}
+        loaded_patch.save(path=patch_location, filesystem=temp_fs)
+        assert not temp_fs.exists(f"{patch_location}/meta_info.json")
+        assert temp_fs.exists(f"{patch_location}/meta_info/beep.json")
+
+        loaded_patch = EOPatch.load(path=patch_location, filesystem=temp_fs)
+        assert dict(loaded_patch.meta_info.items()) == {"beep": "boop"}
```

Comment:
i can re-check the tests if there is a place where I can add more meta-info cases.
But in general I don't want to test too many things of the kind "this file is now here", since the details should kinda be black-boxy. I did white-box testing here, just to make sure that over-saving does not revert back to old-style by accident :scream_cat:

Reply:
Added a test case to feature-partial saving/loading and improved it a bit.
Now it checks that only a single META_INFO feature is loaded at a time (before if one was loaded, they all were)

Any other suggestions?

Reply:
makes sense! cool little change